server/server.js

```diff
@@ -556,6 +556,27 @@ exports.entryPage = "dashboard";
             }
         });
 
+        // Verify if monitor is duplicated
+        socket.on("isDuplicatedMonitor", async (monitor, callback) => {
+            try {
+                let monitorList = await sendMonitorList(socket);
```

Comment:
If you don't check for a login, this is available to everybody.
I think one could extract data via this endpoint if not secured via timing attacks.

While the likelihood of this is low, adding an auth check is probably worth it.
```suggestion
                checkLogin(socket);
                
                let monitorList = await sendMonitorList(socket);
```